src/lib/core/SessionTypes.h

```diff
@@ -0,0 +1,29 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+namespace chip {
+
+enum SessionType : uint8_t
```

Comment:
This is pretty confusing, because a "Session" can be one of: CASE, PASE, group, unauthenticated.

Is this meant to be SecureSessionType?  Or something else?  At a bar minimum this enum needs documentation.

Reply:
Moved this `SecureSession::Type` definition inside `SecureSession` class